### PR TITLE
[Test] add a simple unit tests.

### PR DESCRIPTION
**Pull request by nadavrot**
[Test] add a simple unit tests.

I plan to add new features, such as quantization, to this test in future commits.

### CODE REVIEW

File: tests/unittests/MLTest.cpp [hunk 1]
```diff
@@ -865,6 +865,89 @@ TEST_P(MLTest, convNetForImageRecognition) {
   }
 }
 
+/// Generate data for the regression test. Put a '1' in a random location in a
+/// clear tensor and report the coordinates of that pixel.
+static void generateRegressionTestData(Tensor &images, Tensor &labels) {
+  auto L = labels.getHandle<>();
+  auto image = images.getHandle<>();
+  unsigned numSamples = images.dims()[0];
+  image.clear(-0.5);
```
Comment (rdzhabarov) [inline tests/unittests/MLTest.cpp]: not super clear why -0.5 chosen. Is that somewhat random?

File: tests/unittests/MLTest.cpp [hunk 2]
```diff
@@ -865,6 +865,89 @@ TEST_P(MLTest, convNetForImageRecognition) {
...
+  Function *F = mod.createFunction("main");
+
+  auto *input =
+      mod.createVariable(ElemKind::FloatTy, {batchSize, 10, 10, 1}, "input",
+                         VisibilityKind::Public, Variable::TrainKind::None);
+  auto *ex =
+      mod.createVariable(ElemKind::FloatTy, {batchSize, 2}, "coordinates",
+                         VisibilityKind::Public, Variable::TrainKind::None);
+
+  auto *FC0 = F->createFullyConnected("fc", input, 2);
```
Comment (rdzhabarov) [inline tests/unittests/MLTest.cpp]: is that intentional that there are no hidden layers for the network? (i assume it'd learn better with more layers and we could make that 90% check tighter).

File: tests/unittests/MLTest.cpp [hunk 3]
```diff
@@ -865,6 +865,89 @@ TEST_P(MLTest, convNetForImageRecognition) {
...
+  auto *R = F->createRegression("regression", FC0, ex);
+  auto *result = F->createSave("ret", R);
+  Function *TF = glow::differentiate(F, EE_.getConfig());
+  EE_.compile(CompilationMode::Train, TF);
+
+  Tensor images(ElemKind::FloatTy, {numSamples, 10, 10, 1});
```
Comment (rdzhabarov) [inline tests/unittests/MLTest.cpp]: nit: trainImages and testImages for the images2

**Comment by nadavrot** [comment]
@rdzhabarov I updated the code:
1. Added another layer. 
2. Renamed the variables.
3. Changed -0.5 to zero. 

I kept the 90% hit rate because to get better results we need to train longer, and this makes the unit test run longer.

**Pull request merged**